### PR TITLE
WIP Add Stripe Payment Flow

### PR DESCRIPTION
**Pull request by jtmckibb**

### CODE REVIEW

File: src/components/CheckoutForm/CheckoutForm.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+        <Form.Group as={Row}>
+          <Form.Label column sm="2">
+            Amount
+          </Form.Label>
+          <Col sm="10">
+            <NumberFormat className="form-control" placeholder="e.g. 50.00" decimalScale={2} fixedDecimalScale={true} thousandSeparator={true} allowNegative={false} prefix={'$'} onValueChange={(values) => {
+              const { formattedValue, value } = values
+              const line_items = { ...this.state.line_items, amount: parseFloat(value) }
+              this.setState({ line_items: line_items, formatted_value: formattedValue })
```
Comment (yourfavoritedev) [inline src/components/CheckoutForm/CheckoutForm.tsx:106]: Probably could be its own method `handleOnValueChange`

File: src/components/OwnerPanel/OwnerPanel.tsx [hunk 2]
```diff
@@ -16,8 +17,8 @@ const OwnerPanel: React.SFC<Props> = ({ imageSrc, ownerName, className }) => {
       </figure>
       <h2 className={styles.ownerName}>{ownerName}</h2>
       <div className={styles.buttonContainer}>
-        <button className={classnames(styles.button, "button--filled")}>Donate</button>
-        <button className={classnames(styles.button, "button--outlined")}>Buy Gift Card</button>
+        <PopupModal merchant="Shunfa Bakery" option="Donation" className={classnames(styles.button, "button--filled")} />
```
Comment (jtmckibb) [inline src/components/OwnerPanel/OwnerPanel.tsx:20]: Is this going to change the text in the button from Donate to Donation? We should probably keep the button text at "Donate" and "Buy Gift Card".

**Comment by ting-dd** [comment]
Can you post screenshots?

File: public/index.html [hunk 3]
```diff
@@ -24,6 +24,13 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
+    <link
+      rel="stylesheet"
+      href="https://maxcdn.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css"
+      integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh"
+      crossorigin="anonymous"
+    />
```
[the remark this answers is not in the document]
Comment (karenoujia) [reply public/index.html:32]: Thanks for catching that! I must have done it by accident...I agree that bootstrap is unnecessary for our purposes and we should probably create our own styles. I'm not very familiar with the css rules though but Rob has offered to help with the styling part, so I'm going to remove bootstrap

File: src/components/PopupModal/PopupModal.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+import classnames from 'classnames';
+import styles from './styles.module.scss';
+import Popup from 'reactjs-popup';
```
[the remark this answers is not in the document]
Comment (karenoujia) [reply src/components/PopupModal/PopupModal.tsx:4]: I think we will go with not using bootstrap

File: src/components/CheckoutForm/CheckoutForm.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+            <Form.Control plaintext readOnly defaultValue="USD" />
+          </Col>
+        </Form.Group>
+
+        <Form.Group as={Row}>
+          <Form.Label column sm="2">
+            Amount
+          </Form.Label>
+          <Col sm="10">
+            <NumberFormat className="form-control" placeholder="e.g. 50.00" decimalScale={2} fixedDecimalScale={true} thousandSeparator={true} allowNegative={false} prefix={'$'} onValueChange={(values) => {
```
[the remark this answers is not in the document]
Comment (karenoujia) [reply src/components/CheckoutForm/CheckoutForm.tsx:103]: Intl.NumberFormat is great for formatting but I don't think it controls the precision of the value entered (we should probably automatically display the amount to 2 decimal places) so I'm going to leave this library here for now

File: src/components/CheckoutForm/CheckoutForm.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+    axios.post('https://whispering-springs-34358.herokuapp.com/', {
+      line_items: [line_items],
+      merchant_id: this.state.merchant_id
+    }, {
+    	headers: {
+    	  'Access-Control-Allow-Origin': '*',
+      }
+    })
+    .then(res => console.log(res))
```
[the remark this answers is not in the document]
Comment (karenoujia) [reply src/components/CheckoutForm/CheckoutForm.tsx:74]: this gets the stripe checkout session to start the payment flow

**Comment by karenoujia** [comment]
modal_screenshot
This is what it looks like after removing bootstrap - clicking on either the "Donate" or "Buy Gift Card" button brings up the modal, and the input fields reflect what's needed by the backend to create a new checkout session. The next step is to use the returned checkout id to begin the payment process. Still need some help styling the form and the modal (which currently has some template css I pulled from reactjs-popup).

**Comment by jtmckibb** [comment]
Looks good to merge, and we'll let @ArtyEmsee iterate on it

**Pull request merged**